src/bio/bio_xstream.c

```diff
@@ -160,6 +210,22 @@ populate_whitelist(struct spdk_env_opts *opts)
 			break;
 		}
 
+		if (vmd_enabled) {
+			/*
+			 * TODO: Remove duplicate VMD addr being added to
+			 * whitelist for multiple NVMe SSDs behind the same VMD.
+			 */
```

Comment:
this is still a TODO. I can try to upload a solution for this, not sure if you had any ideas about this

Reply:
I'm not sure about making the assumption that a nonzero beginning to a PCI address is always going to be a backing address but maybe it's okay for the moment. Probably the best solution for duplicate whitelist entries is to make `opts_add_pci_addr()` ignore duplicates, I will make that change.

Reply:
I think the assumption of a prefix of 1000* for VMD will work instead. yes we need to ignore duplicates, I wasn't sure the most efficient way to do this. sounds like you might have an idea.

Reply:
again , please check the most recent commits to see if that solution looks okay, let's leave the prefix for the moment as we have bigger fish to fry